### PR TITLE
Optimisations and Fixes

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

File: Common/src/gRPC/Services/Submitter.cs [hunk 1]
```diff
@@ -276,9 +281,18 @@ await foreach (var taskRequest in taskRequests.WithCancellation(cancellationToke
...
+    activity?.AddTag("sessionId",
+                     session);
+    activity?.AddTag("parentTaskId",
+                     parentTaskId);
+    activity?.AddTag("dispatchId",
+                     dispatchId);
+    activity?.AddTag("taskIds",
+      string.Join(",", requests.Select(request => request.Id)));
+
+    // todo: clean
```
Comment (jfonseca-aneo) [inline Common/src/gRPC/Services/Submitter.cs:295]: Can you add more details to this todo?

File: Tests/HtcMock/Client/src/GridClient.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        Id = sessionId,
+      };
+      var session = client_.CreateSession(createSessionRequest);
+      switch (session.ResultCase)
+      {
+        case CreateSessionReply.ResultOneofCase.Error:
+          throw new Exception("Error while creating session : " + session.Error);
+        case CreateSessionReply.ResultOneofCase.None:
+          throw new Exception("Issue with Server !");
+        case CreateSessionReply.ResultOneofCase.Ok:
```
Comment (jfonseca-aneo) [inline Tests/HtcMock/Client/src/GridClient.cs:74]: a debug log here printing the session id may be useful if in the future a problem within that session arrises.

File: Common/src/gRPC/Services/Submitter.cs (hunk 1 above)
Comment (aneojgurhem) [reply Common/src/gRPC/Services/Submitter.cs:295]: done !

File: Tests/HtcMock/Client/src/GridClient.cs (hunk 2 above)
Comment (aneojgurhem) [reply Tests/HtcMock/Client/src/GridClient.cs:74]: added

**Pull request merged**